Content.Server/Xenoarchaeology/Artifact/XAE/Components/XAECreatePuddleComponent.cs

```diff
@@ -0,0 +1,37 @@
+using Content.Shared.Chemistry.Components;
+using Content.Shared.Chemistry.Reagent;
+using Robust.Shared.Prototypes;
+
+namespace Content.Server.Xenoarchaeology.Artifact.XAE.Components;
+
+/// <summary>
+/// This is used for an artifact that creates a puddle of
+/// random chemicals upon being triggered.
+/// </summary>
+[RegisterComponent, Access(typeof(XAECreatePuddleSystem))]
+public sealed partial class XAECreatePuddleComponent : Component
+{
+    /// <summary>
+    /// The solution where all the chemicals are stored
+    /// </summary>
+    [DataField(required: true), ViewVariables(VVAccess.ReadWrite)]
+    public Solution ChemicalSolution = default!;
+
+    /// <summary>
+    /// The different chemicals that can be spawned by this effect
+    /// </summary>
+    [DataField]
+    public List<ProtoId<ReagentPrototype>> PossibleChemicals = default!;
+
+    /// <summary>
+    /// The number of chemicals in the puddle
+    /// </summary>
+    [DataField]
+    public int ChemAmount = 3;
```

Comment:
Effects code was my doing, not Emo, and i was transferring code 'as it was' most of the time (because ther was A LOT of it). I'd say that it 100% can be min-max, and i will fix it in this pr, yes.